### PR TITLE
Update protocols to match latest core schema

### PR DESCRIPTION
**Pull request by dgarros**
Related to https://github.com/opsmill/infrahub/pull/4716

### CODE REVIEW

File: infrahub_sdk/protocols.py [hunk 1]
```diff
@@ -397,7 +399,6 @@ class CoreNumberPool(CoreResourcePool, LineageSource):
 
 
 class CoreObjectPermission(CoreBasePermission):
-    branch: String
```
Comment (dgarros) [inline infrahub_sdk/protocols.py:400]: @gmazoyer is this change expected ? it wasn't part of my change

Comment (ogenstad) [reply infrahub_sdk/protocols.py:400]: That should be fine as we decided to remove the "branch" parts as part of the object permissions with the argument that users probably wouldn't want to setup permissions on specific branches and instead have settings for "allow on default branch" or "allow on other branches" on in the future "allow on branches that I have created".

**Review by ogenstad — APPROVED** [review]

File: infrahub_sdk/protocols.py [hunk 2]
```diff
@@ -7,7 +7,8 @@
 from .protocols_base import CoreNode, CoreNodeSync
 
 if TYPE_CHECKING:
-    from .node import RelatedNode, RelatedNodeSync, RelationshipManager, RelationshipManagerSync
+    from infrahub_sdk.node import RelatedNode, RelatedNodeSync, RelationshipManager, RelationshipManagerSync
```
Comment (ogenstad) [inline infrahub_sdk/protocols.py:10]: Was there some problem with the relative import here?

Comment (dgarros) [reply infrahub_sdk/protocols.py:10]: I haven't touched this part, not sure why the import was relative in the first place
this is what the current code is generating

**Pull request merged**